pkg/config/config.go

```diff
@@ -36,6 +37,38 @@ func LoadConf(bytes []byte) (*sriovtypes.NetConf, error) {
 		return nil, fmt.Errorf("LoadConf(): VF pci addr is required")
 	}
 
+	// Get the vf pci address
+	vfPciAddress, err := utils.GetPciAddress(n.Master, n.VFID)
+	if err != nil {
+		return n, fmt.Errorf("error getting vf pci address for PF %s and VF id %d: %v", n.Master, n.VFID, err)
+	}
+
+	n.VFPciAddress = vfPciAddress
+
+	// Check if the device is already allocated.
+	// This is to prevent issues where kubelet request to delete a pod and in the same time a new pod using the same
+	// vf is started. we can have an issue where the cmdDel of the old pod is called AFTER the cmdAdd of the new one
+	// This will block the new pod creation until the cmdDel is done.
+	isAllocated, allocatedNs, err := utils.IsAllocated(vfPciAddress, DefaultCNIDir)
+	if err != nil {
+		return n, err
+	}
+
+	if isAllocated {
```

Comment:
I moved the verification from the main to here so we produce a better error when a PCI address is still assign.
instead of this error

```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_client_seba_65fe3641-ebd8-4f8b-9b17-785f07b3ad3b_0(0140bb9890307e2f3f8804a06dd0f70a0970f7b8ee1254227eb5c0043ddf2f77): error adding pod seba_client to CNI network "multus-cni-network": plugin type="multus" name="multus-cni-network" failed (add): [seba/client/65fe3641-ebd8-4f8b-9b17-785f07b3ad3b:sriov-network-2]: error adding container to network "sriov-network-2": SRIOV-CNI failed to load netconf: LoadConf(): the VF 0000:d8:0b.0 does not have a interface name or a dpdk driver
```

we will return error that the PCI is still allocated